src/intersperse.rs

```diff
@@ -54,18 +54,17 @@ where
 {
     element: ElemF,
     iter: Fuse<I>,
-    peek: Option<I::Item>,
+    peek: Option<Option<I::Item>>,
```

Comment:
With the added nesting, could you just write a comment here about how the value of `peek` should be interpreted? (`None`, `Some(None)`, `Some(Some(...))`?